### PR TITLE
Allow run_app(access_log=None)

### PR DESCRIPTION
**Pull request by hynek**
## What do these changes do?

The docs suggest that you can disable request logging by passing `access_log=None` to run_app.

However:

1. It breaks when loop's debug is True
2. The type stubs don't reflect that

This PR fixes both.

## Are there changes in behavior for the user?

Documented behavior works under more conditions now. :)

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: CHANGES/3504.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix type stubs for ``aiohttp.web.run_app(access_log=True)`` and fix edge case of ``access_log=True?? and the event loop being in debug mode.
```
Comment (kxepal) [inline CHANGES/3504.bugfix]: `??` need to replace with closing back quotes.

Comment (hynek) [reply CHANGES/3504.bugfix]: oh yeah sorry, notebook keyboard :)

File: aiohttp/web.py [hunk 2]
```diff
@@ -383,15 +383,15 @@ def run_app(app: Union[Application, Awaitable[Application]], *,
...
+            access_log: Optional[logging.Logger]=access_logger,
             handle_signals: bool=True,
             reuse_address: Optional[bool]=None,
             reuse_port: Optional[bool]=None) -> None:
     """Run an app locally"""
     loop = asyncio.get_event_loop()
 
     # Configure if and only if in debugging mode and using the default logger
-    if loop.get_debug() and access_log.name == 'aiohttp.access':
+    if loop.get_debug() and access_log and access_log.name == 'aiohttp.access':
```
Comment (webknjaz) [inline aiohttp/web.py:394]: maybe
```suggestion
    if loop.get_debug() and getattr(access_log, 'name', None) == 'aiohttp.access':
```

File: aiohttp/web.py [hunk 3]
```diff
@@ -269,7 +269,7 @@
                    backlog: int=128,
                    access_log_class: Type[AbstractAccessLogger]=AccessLogger,
                    access_log_format: str=AccessLogger.LOG_FORMAT,
-                   access_log: logging.Logger=access_logger,
+                   access_log: Optional[logging.Logger]=access_logger,
```
Comment (Arfey) [inline aiohttp/web.py:272]: why do u use `Optional`, and when access_log should be None?

Comment (Arfey) [reply aiohttp/web.py:272]: i just don't understand when u need to drop logging :)

Comment (hynek) [reply aiohttp/web.py:272]: Firstly, this is just an adoption of the stubs to reflect existing [docs](https://docs.aiohttp.org/en/stable/logging.html#access-logs), secondly I ask myself the vice versa question every time I try to get rid of some framework's logging. I have “dumb” HTTP logs in my load balancer and I want my applications to have only structured application logs.

Comment (Arfey) [reply aiohttp/web.py:272]: The load balancer is story about production, right? do u use debug mode for production?
Now, u can put your custom logger and this should solve your problem.

Comment (hynek) [reply aiohttp/web.py:272]: This isn't about *my* problems, this is about making the code do what the docs say. :)

Comment (Arfey) [reply aiohttp/web.py:272]: so maybe the problem is not in the code and in the documentation :)
I don't find any test for this behavior and this feature never worked. I am not sure that this feature is required.

Comment (hynek) [reply aiohttp/web.py:272]: It worked when debug was off which is the normal case. I ran into the edge case of debug=True thanks to mypy.

Comment (Arfey) [reply aiohttp/web.py:272]: You are right without mypy all is fine. I think will be cool have some test for this case.

**Comment by asvetlov** [comment]
Thanks!
